internal/metricstestutils/metrics_testutils.go

```diff
@@ -0,0 +1,159 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metricstestutils
+
+import (
+	"time"
+
+	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+	timestamppb "github.com/golang/protobuf/ptypes/timestamp"
+	wrapperspb "github.com/golang/protobuf/ptypes/wrappers"
+)
+
+// Gauge creates a gauge metric.
+func Gauge(name string, keys []string, timeseries ...*metricspb.TimeSeries) *metricspb.Metric {
+	return metric(metricspb.MetricDescriptor_GAUGE_DOUBLE, name, keys, timeseries)
+}
+
+// GaugeInt creates a gauge metric of type int64.
+func GaugeInt(name string, keys []string, timeseries ...*metricspb.TimeSeries) *metricspb.Metric {
+	return metric(metricspb.MetricDescriptor_GAUGE_INT64, name, keys, timeseries)
+}
+
+// GaugeDist creates a gauge distribution metric.
+func GaugeDist(name string, keys []string, timeseries ...*metricspb.TimeSeries) *metricspb.Metric {
+	return metric(metricspb.MetricDescriptor_GAUGE_DISTRIBUTION, name, keys, timeseries)
+}
+
+// Cumulative creates a cumulative metric.
+func Cumulative(name string, keys []string, timeseries ...*metricspb.TimeSeries) *metricspb.Metric {
+	return metric(metricspb.MetricDescriptor_CUMULATIVE_DOUBLE, name, keys, timeseries)
+}
+
+// CumulativeInt creates a cumulative metric of type int64.
+func CumulativeInt(name string, keys []string, timeseries ...*metricspb.TimeSeries) *metricspb.Metric {
+	return metric(metricspb.MetricDescriptor_CUMULATIVE_INT64, name, keys, timeseries)
+}
+
+// CumulativeDist creates a cumulative distribution metric.
+func CumulativeDist(name string, keys []string, timeseries ...*metricspb.TimeSeries) *metricspb.Metric {
+	return metric(metricspb.MetricDescriptor_CUMULATIVE_DISTRIBUTION, name, keys, timeseries)
+}
+
+// Summary creates a summary metric.
+func Summary(name string, keys []string, timeseries ...*metricspb.TimeSeries) *metricspb.Metric {
+	return metric(metricspb.MetricDescriptor_SUMMARY, name, keys, timeseries)
+}
+
+// Timeseries creates a timeseries. It takes the start time stamp, a sequence of label values (associated
+// with the label keys in the overall metric), and the value of the timeseries.
+func Timeseries(sts time.Time, vals []string, point *metricspb.Point) *metricspb.TimeSeries {
+	return &metricspb.TimeSeries{
+		StartTimestamp: ToTS(sts),
+		Points:         []*metricspb.Point{point},
+		LabelValues:    toVals(vals),
+	}
+}
+
+// Double creates a double point.
+func Double(ts time.Time, value float64) *metricspb.Point {
+	return &metricspb.Point{Timestamp: ToTS(ts), Value: &metricspb.Point_DoubleValue{DoubleValue: value}}
+}
+
+// Dist creates a distribution point. It takes the time stamp, the bucket boundaries for the distribution, and
+// the and counts for the individual buckets as input.
+func Dist(ts time.Time, bounds []float64, counts []int64) *metricspb.Point {
```

Comment:
When this package is moved out of internal (as suggested in another comment), it would be better to use DistributionPoint instead of Dist.

Reply:
I don't want to make the function names to be to verbose but I think I see your point. I've updated the name to DistPt for clarity but we can revisit the name when we move it out of internal.